examples/AdvancedOperations/AddAppCampaign.php

```diff
@@ -0,0 +1,443 @@
+            // corresponding in-app actions.
+            // 'selective_optimization' => new SelectiveOptimization([
+            //     'conversion_actions' => [
+            //         new StringValue(['value' => 'INSERT_CONVERSION_TYPE_ID_HERE'])
+            //     ]
+            // ])
+        ]);
+
+        // Creates a campaign operation.
+        $campaignOperation = new CampaignOperation();
+        $campaignOperation->setCreate($campaign);
+
+        // Submits the campaign operation and prints the results.
+        $campaignServiceClient = $googleAdsClient->getCampaignServiceClient();
+        $response = $campaignServiceClient->mutateCampaigns($customerId, [$campaignOperation]);
+
+        $createdCampaignResourceName = $response->getResults()[0]->getResourceName();
+        printf("Created App campaign '%s'%s", $createdCampaignResourceName, PHP_EOL);
+
+        return $createdCampaignResourceName;
+    }
+
+    /**
+     * Sets campaign targeting criteria for a given campaign.
+     *
+     * Both location and language targeting are illustrated.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @param string $campaignResourceName the resource name of the campaign to apply targeting to
+     */
+    private static function setCampaignTargetingCriteria(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $campaignResourceName
+    ) {
+        $campaignCriterionOperations = [];
+
+        // Creates the location campaign criteria.
+        // Besides using location_id, you can also search by location names from
+        // GeoTargetConstantService.suggestGeoTargetConstants() and directly
+        // apply GeoTargetConstant.resource_name here. An example can be found
+        // in GetGeoTargetConstantByNames.php.
+        $locationIds = [
+            21137, // California
+            2484   // Mexico
+        ];
+        foreach ($locationIds as $locationId) {
+            // Creates a campaign criterion.
+            $campaignCriterion = new CampaignCriterion([
+                'campaign' => new StringValue(['value' => $campaignResourceName]),
+                'type' => CriterionType::LOCATION,
+                'location' => new LocationInfo([
+                    'geo_target_constant' => new StringValue([
+                        'value' => ResourceNames::forGeoTargetConstant($locationId)
+                    ])
+                ])
+            ]);
+
+            // Creates a campaign criterion operation.
+            $campaignCriterionOperation = new CampaignCriterionOperation();
+            $campaignCriterionOperation->setCreate($campaignCriterion);
+
+            $campaignCriterionOperations[] = $campaignCriterionOperation;
+        }
+
+        // Creates the language campaign criteria.
+        $languageIds = [
+            1000, // English
```

Comment:
Probably it's fine, but could you double check if this style complies with PSR-2?

Reply:
PHPCS runs without errors/warnings